source/write/transactions.txt

```diff
@@ -0,0 +1,170 @@
+the session state by using the methods provided by the returned ``ClientSession``.
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``start_transaction()``
+     - | Starts a new transaction, configured with the given options, on
+         this session. Returns an error if there is already
+         a transaction in progress for the session. To learn more about
+         this method, see the :manual:`startTransaction() page
+         </reference/method/Session.startTransaction/>`  in the Server manual.
+       |
+       | **Parameters**: ``read_concern``, ``write_concern``, ``read_preference``, ``max_commit_time_ms``
+       | **Return Type**: ``ContextManager``
+
+   * - ``abort_transaction()``
+     - | Ends the active transaction for this session. Returns an
+         error if there is no active transaction for the session or the
+         transaction has been committed or ended. To learn more about
+         this method, see the :manual:`abortTransaction() page
+         </reference/method/Session.abortTransaction/>`  in the Server manual.
+       |
+
+   * - ``commit_transaction()``
+     - | Commits the active transaction for this session. Returns an
+         error if there is no active transaction for the session or if the
+         transaction was ended. To learn more about
+         this method, see the :manual:`commitTransaction() page
+         </reference/method/Session.commitTransaction/>`  in the Server manual.
+
+   * - ``with_transaction()``
+     - | Starts a transaction on this session and executes a callback once, then
+         commits the transaction. In the event of an exception, this method may retry
+         the commit or the entire transaction, which may invoke the callback multiple 
+         times by a single call to ``with_transaction``.
+       |
+       | **Parameters**: ``callback``, ``read_concern``, ``write_concern``, ``read_preference``, ``max_commit_time_ms``
+       | **Return Type**: ``_T``
+
+   * - ``end_session()``
+     - | Finishes this session. If a transaction has started, this method aborts it.
+         Returns an error if there is no active session to end.
+
+A ``ClientSession`` also has methods to retrieve session
+properties and modify mutable session properties. View the :ref:`API
+documentation <api-docs-transaction>` to learn more about these methods.
+
+Example
+-------
+
+The following example shows how you can create a session, create a
+transaction, and commit a multi-document insert operation through the
+following steps:
+
+1. Create a session from the client using the ``start_session()`` method.
+#. Use the ``with_transaction()`` method to start a transaction.
+#. Insert multiple documents. The ``with_transaction()`` method executes the
+   insert and commits the transaction. If any operation results in
+   errors, ``with_transaction()`` handles canceling the transaction. This method
```

Comment:
S: to [avoid "execute"](https://www.mongodb.com/docs/meta/style-guide/terminology/terms-for-global-audience/#use-jargon-with-care) and clarify/condense a little:
```suggestion
#. Insert multiple documents. The ``with_transaction()`` method runs the
   insert operation and commits the transaction. If any operation results in
   errors, ``with_transaction()`` cancels the transaction. This method
```